src/storage/sqlite.rs

```diff
@@ -368,6 +461,160 @@ impl StorageRaw for SqliteStorage {
+        let mut info = StorageTokenInfo::default();
+        let conn = self.conn.lock()?;
+        let mut stmt = conn
+            .prepare("SELECT value, data from meta WHERE name='TOKEN INFO'")?;
+        let mut rows = stmt.query(params![])?;
+        while let Some(row) = rows.next()? {
+            let name: String = row.get(0)?;
+            let value: Vec<u8> = row.get(1)?;
+            match name.as_str() {
+                "label" => copy_sized_string(value.as_slice(), &mut info.label),
+                "manufacturer" => {
+                    copy_sized_string(value.as_slice(), &mut info.manufacturer)
+                }
+                "model" => copy_sized_string(value.as_slice(), &mut info.model),
+                "serial" => {
+                    copy_sized_string(value.as_slice(), &mut info.serial)
+                }
+                "flags" => {
+                    info.flags = CK_FLAGS::try_from(u32::from_le_bytes(
+                        value.try_into()?,
+                    ))?
+                }
+                _ => (), /* ignore unknown values for forward compatibility */
+            }
+        }
+        Ok(info)
+    }
+
+    fn store_token_info(&mut self, info: &StorageTokenInfo) -> Result<()> {
+        let mut conn = self.conn.lock()?;
+        let mut tx = conn.transaction().map_err(bad_storage)?;
+        tx.set_drop_behavior(rusqlite::DropBehavior::Rollback);
+
+        /* First delete current rows, then INSERT new */
+        //Self::delete_meta(&mut tx, "TOKEN INFO", None, None, None)?;
+        Self::store_meta(
+            &mut tx,
+            "TOKEN INFO",
+            None,
+            Some("label"),
+            Some(&info.label as &[u8]),
+        )?;
+        Self::store_meta(
+            &mut tx,
+            "TOKEN INFO",
+            None,
+            Some("manufacturer"),
+            Some(&info.manufacturer as &[u8]),
+        )?;
+        Self::store_meta(
+            &mut tx,
+            "TOKEN INFO",
+            None,
+            Some("model"),
+            Some(&info.model as &[u8]),
+        )?;
+        Self::store_meta(
+            &mut tx,
+            "TOKEN INFO",
+            None,
+            Some("serial"),
+            Some(&info.serial as &[u8]),
+        )?;
+        /* filter out runtime flags */
+        let flags = info.flags
+            & (CKF_LOGIN_REQUIRED
+                | CKF_TOKEN_INITIALIZED
+                | CKF_WRITE_PROTECTED);
+        let flags32 = u32::try_from(flags)?;
+        Self::store_meta(
+            &mut tx,
+            "TOKEN INFO",
+            None,
+            Some("flags"),
+            Some(&flags32.to_le_bytes() as &[u8]),
+        )?;
+
+        tx.commit().map_err(bad_storage)
+    }
+
+    fn fetch_user(&self, uid: &str) -> Result<StorageAuthInfo> {
+        let conn = self.conn.lock()?;
+        let mut stmt = conn
+            .prepare("SELECT name, data from meta WHERE ((name=? OR name=? OR name=?) AND value=?)")?;
```

Comment:
`name in (?, ?, ?)` SQL would be more appropriate here:

https://www.techonthenet.com/sqlite/in.php

Not sure if the rustqlite have some even better shorthands for the construction of list to make it even simpler.

Reply:
no only has parameters, it is a very plain shim on top if libsqlite3